### PR TITLE
Campaign list on index page

### PR DESCRIPTION
**Pull request by schneiderman805**
## PT: 
[Link to PT](https://www.pivotaltracker.com/story/show/162396962)

## User story:
```
As a User
In order to know which campaigns I can donate to
I would like to see a list of available campaigns on the index page
```
## Actions taken:

- Adds controller model and views for index page

- Adds validation 

## Lessons learned:

- Watch out for typos
screenshot 2018-12-04 at 19 28 05

### CODE REVIEW

**Review by CamronLDNF — COMMENTED** [review]
As discussed, let's move the image attribute from the campaign cards out to its own user story and for later date to implement.

**Review by HannaSwDn — APPROVED** [review]

File: app/models/campaign.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Campaign < ApplicationRecord
+    validates :title, presence: true
+    validates :description, presence: true
+    validates :amount_raised, presence: true
```
Comment (faraznaeem) [inline app/models/campaign.rb]: `validates_presence_of :title, :description, :amount_raised`

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'campaigns/index'
```
Comment (faraznaeem) [inline config/routes.rb]: remove the route
please use `--skip-routes` when you are using the controller generate command

File: features/User_can_see_the_campaigns_on_Index_page.feature [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    As a User
+    In order to know which campaigns I can donate to
+    I would like to see a list of available campaigns on the index page
+    
+    Scenario: User can see the campaigns on index page
+        Given the following campaigns exists
+        | title        | description                      | amount_raised |
+        |Brain surgery | I grew up always helping others. | 1000.00       |
+     
+        When I visit the site
```
Comment (faraznaeem) [inline features/User_can_see_the_campaigns_on_Index_page.feature]: `And i visit the site`

File: spec/modles/campaigns_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
```
Comment (faraznaeem) [inline spec/modles/campaigns_spec.rb]: `spec/modles/campaigns_spec.rb` ???
 why didn't you use the scaffolded file?

File: app/assets/stylesheets/campaigns.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Campaigns controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (faraznaeem) [inline app/assets/stylesheets/campaigns.scss:3]: Why are these files created, aren't the generators not turned off?

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/views/campaigns/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+%h1 List of Campaigns
+- @campaigns.each do |campaign|
+    %p= campaign.title
+    %p= campaign.description
+    %p= campaign.amount_raised
```
Comment (faraznaeem) [inline app/views/campaigns/index.html.haml:5]: please add a screenshot

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/models/campaign.rb (hunk 1 above)
Comment (schneiderman805) [reply app/models/campaign.rb]: we have refactored code into one line

File: config/routes.rb (hunk 2 above)
Comment (schneiderman805) [reply config/routes.rb]: we have removed the route

File: features/User_can_see_the_campaigns_on_Index_page.feature (hunk 3 above)
Comment (schneiderman805) [reply features/User_can_see_the_campaigns_on_Index_page.feature]: we have changed When to And

File: spec/modles/campaigns_spec.rb (hunk 4 above)
Comment (schneiderman805) [reply spec/modles/campaigns_spec.rb]: we had duplicate files so we removed one

File: app/assets/stylesheets/campaigns.scss (hunk 5 above)
Comment (schneiderman805) [reply app/assets/stylesheets/campaigns.scss:3]: we are not quite sure, we could possible use a little explaining please.

File: app/views/campaigns/index.html.haml (hunk 6 above)
Comment (schneiderman805) [reply app/views/campaigns/index.html.haml:5]: we have taken a screen shot and added it to this PR.

**Review by HannaSwDn — APPROVED** [review]

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**